pkg/disttask/framework/storage/task_table.go

```diff
@@ -687,35 +671,53 @@ func (stm *TaskManager) HasSubtasksInStates(ctx context.Context, tidbID string,
 }
 
 // StartSubtask updates the subtask state to running.
-func (stm *TaskManager) StartSubtask(ctx context.Context, subtaskID int64) error {
-	_, err := stm.executeSQLWithNewSession(ctx, `update mysql.tidb_background_subtask
-		set state = %?, start_time = unix_timestamp(), state_update_time = unix_timestamp()
-		where id = %?`,
-		proto.TaskStateRunning, subtaskID)
-	return err
+func (stm *TaskManager) StartSubtask(ctx context.Context, subtaskID int64, execID string) (bool, error) {
+	// owned means the task executor can own and run the subtask without any other task executor running it.
```

Comment:
maybe define a named err as `SubtaskNotFound`？ to mean that the subtask of the id and exec_id is not found

Reply:
IMHO, notfound subtask is not an error, if use error may change some error handling code.

Reply:
Notfound errors are very common, I prefer using "error" more. We seem don't need handling this error specifically

<img width="663" alt="image" src="https://github.com/pingcap/tidb/assets/23556834/242e02ce-e64a-4dba-898d-69c0b19f7269">

Reply:
> Notfound errors are very common, I prefer using "error" more. We seem don't need handling this error specifically
> 
> <img alt="image" width="663" src="https://private-user-images.githubusercontent.com/23556834/292948436-242e02ce-e64a-4dba-898d-69c0b19f7269.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NTAxOTMsIm5iZiI6MTcwMzY0OTg5MywicGF0aCI6Ii8yMzU1NjgzNC8yOTI5NDg0MzYtMjQyZTAyY2UtZTY0YS00ZGJhLTg5OGQtNjljMGIxOWY3MjY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDA0MDQ1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY5MTBlNTVlYzQzYmQ4YTdjNTViOTlhMDZlNWU3MzViMWYwMWMzNWZhZTY0NWQyMWJkYmYyOWYxY2U3OGJkYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GhhA7QmS5n7wMZJOWJUDdpq8IT-rR9OxuVue72VIERk">

Indeed we need to filter out NotFound error and do not mark the subtasks on task executor as failed.
See logErrAndPersist